src/program/mod.rs

```diff
@@ -283,9 +283,108 @@ pub enum ProgramCreationInput<'a> {
 
         /// Whether the shader uses point size.
         uses_point_size: bool,
+    },
+
+    /// Use a SPIR-V binary.
+    SpirV(SpirvProgram<'a>),
+}
+
+/// Represents a SPIR-V program. The shaders can refer to entry points in the same binary.
+#[derive(Clone)]
+pub struct SpirvProgram<'a> {
+    /// The vertex shader.
+    pub vertex_shader: SpirvEntryPoint<'a>,
+
+    /// The fragment shader.
+    pub fragment_shader: SpirvEntryPoint<'a>,
+
+    /// Optional tessellation control shader.
+    pub tessellation_control_shader: Option<SpirvEntryPoint<'a>>,
+
+    /// Optional tessellation evaluation shader.
+    pub tessellation_evaluation_shader: Option<SpirvEntryPoint<'a>>,
+
+    /// Optional geometry shader.
+    pub geometry_shader: Option<SpirvEntryPoint<'a>>,
+
+    /// The list of variables and mode to use for transform feedback.
+    ///
+    /// The information specified here will be passed to the OpenGL linker. If you pass
+    /// `None`, then you won't be able to use transform feedback.
+    pub transform_feedback_varyings: Option<(Vec<String>, TransformFeedbackMode)>,
+
+    /// See `SourceCode::outputs_srgb`.
+    pub outputs_srgb: bool,
+
+    /// Whether the shader uses point size.
+    pub uses_point_size: bool,
+}
+
+impl<'a> SpirvProgram<'a> {
+    /// Create new `SpirvProgram` from vertex and fragment shaders.
+    pub fn from_vs_and_fs(
+        vertex_shader: SpirvEntryPoint<'a>,
+        fragment_shader: SpirvEntryPoint<'a>,
+    ) -> Self {
+        Self {
+            vertex_shader,
+            fragment_shader,
+            tessellation_control_shader: Default::default(),
+            tessellation_evaluation_shader: Default::default(),
+            geometry_shader: Default::default(),
+            transform_feedback_varyings: Default::default(),
+            outputs_srgb: Default::default(),
+            uses_point_size: Default::default(),
```

Comment:
Could you add false, None, etc instead of Default? This would make this code more informative.

Reply:
Sure. Done.